e2e/e2etests/test_precompiles_distribute_through_contract.go

```diff
@@ -0,0 +1,122 @@
+package e2etests
+
+import (
+	"math/big"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/e2e/contracts/testdistribute"
+	"github.com/zeta-chain/node/e2e/runner"
+	"github.com/zeta-chain/node/e2e/utils"
+	"github.com/zeta-chain/node/precompiles/bank"
+	"github.com/zeta-chain/node/precompiles/staking"
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+)
+
+func TestPrecompilesDistributeThroughContract(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 0, "No arguments expected")
+
+	var (
+		spenderAddress            = r.EVMAddress()
+		distributeContractAddress = staking.ContractAddress
+		feeCollectorAddress       = authtypes.NewModuleAddress("fee_collector")
```

Comment:
Consider adding this address into the runner

Reply:
Added in Modified in 06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf